### ISSUE TITLE
Rapp index builder cannot locate all rapp resource files

### ISSUE THREAD

**Issue by bit-pirate** [robotics-in-concert/rocon_app_platform#253]
The rapp index building feature of the `rocon_app` tool has problems locating all required resource files. This might has to do something with the recent upgrade to using relative paths for rapp resources.

@jihoonl knows more details about this recent upgrade.

@dirk-thomas could you update the indexer?

**Comment by jihoonl** [comment]
I was trying to index [rocon_apps package](https://github.com/robotics-in-concert/rocon_app_platform/tree/indigo/rocon_apps) and found out it does not index resource files correctly.

It should have include icon, launch, parameter, and interface resource path. But it only contains .rapp file

```
  1 " tar.vim version v29
  2 " Browsing tarfile /home/jihoonl/research/ros/rocon/src/rocon_app_platform/rocon_apps.index.tar.gz
  3 " Select a file with cursor and press ENTER
  4 
  5 rocon_apps/package.xml
  6 rocon_apps/chirp_apps/meow/meow_chirp.rapp
  7 rocon_apps/apps/chirp/chirp.rapp
  8 rocon_apps/apps/teleop/teleop.rapp
  9 rocon_apps/chirp_apps/moo/moo_chirp.rapp
 10 rocon_apps/apps/video_teleop/video_teleop.rapp
 11 rocon_apps/apps/talker/talker.rapp
 12 rocon_apps/apps/listener/listener.rapp
 13 rocon_apps/chirp_apps/angry_cat/angry_cat_chirp.rapp
 14 rocon_apps/chirp_apps/lion/lion_chirp.rapp
```

**Comment by jihoonl** [comment]
So I added more debug message on 'write_tarball` for introspection like in [this](https://github.com/robotics-in-concert/rocon_app_platform/blob/rapp_indexer_bugfix/rocon_app_utilities/src/rocon_app_utilities/indexer.py#L273-L310) in  rapp_indexer_bugfix branch.

And this is output

```
write_tarball() to 'rocon_apps...'
write_tarball() add package.xml 'rocon_apps/package.xml
write_tarball() add .rapp file 'rocon_apps/chirp_apps/meow/meow_chirp.rapp
write_index() path does not exist rocon_apps/chirp_apps/meow/rocon_apps/chirp
write_index() path does not exist rocon_apps/chirp_apps/meow/Make an audible "meow" sound.
write_index() path does not exist rocon_apps/chirp_apps/meow/Meow Chirp
write_index() path does not exist rocon_apps/chirp_apps/meow/rocon:/
write_index() path does not exist rocon_apps/chirp_apps/meow/rocon_apps/chirp_apps/meow/meow_chirp.launch
write_tarball() add .rapp file 'rocon_apps/apps/chirp/chirp.rapp
write_index() path does not exist rocon_apps/apps/chirp/Make an audible "chirp" sound.
write_index() attribute error : 'dict' object has no attribute 'startswith'
write_index() path does not exist rocon_apps/apps/chirp/Chirp
write_index() path does not exist rocon_apps/apps/chirp/rocon_apps/icons/cow.png
write_tarball() add .rapp file 'rocon_apps/apps/teleop/teleop.rapp
write_index() path does not exist rocon_apps/apps/teleop/Teleoperate ...
write_index() attribute error : 'dict' object has no attribute 'startswith'
write_index() path does not exist rocon_apps/apps/teleop/Teleop
write_index() path does not exist rocon_apps/apps/teleop/rocon_apps/apps/teleop/joystick.png
write_tarball() add .rapp file 'rocon_apps/chirp_apps/moo/moo_chirp.rapp
write_index() path does not exist rocon_apps/chirp_apps/moo/rocon_apps/chirp
write_index() path does not exist rocon_apps/chirp_apps/moo/Make an audible "moo" sound.
write_index() path does not exist rocon_apps/chirp_apps/moo/Moo Chirp
write_index() path does not exist rocon_apps/chirp_apps/moo/rocon:/
write_index() path does not exist rocon_apps/chirp_apps/moo/rocon_apps/chirp_apps/moo/moo_chirp.launch
write_tarball() add .rapp file 'rocon_apps/apps/video_teleop/video_teleop.rapp
write_index() path does not exist rocon_apps/apps/video_teleop/Teleoperate with image stream...
write_index() attribute error : 'dict' object has no attribute 'startswith'
write_index() path does not exist rocon_apps/apps/video_teleop/VideoTeleop
write_index() path does not exist rocon_apps/apps/video_teleop/rocon_apps/apps/video_teleop/video_teleop.png
write_tarball() add .rapp file 'rocon_apps/apps/talker/talker.rapp
write_index() path does not exist rocon_apps/apps/talker/Default ros style talker tutorial
write_index() path does not exist rocon_apps/apps/talker/rocon_apps/icons/conductor.png
write_index() path does not exist rocon_apps/apps/talker/rocon_apps/apps/talker/talker.launch
write_index() path does not exist rocon_apps/apps/talker/rocon:/
write_index() attribute error : 'dict' object has no attribute 'startswith'
write_index() path does not exist rocon_apps/apps/talker/Talker
write_index() attribute error : 'dict' object has no attribute 'startswith'
write_tarball() add .rapp file 'rocon_apps/apps/listener/listener.rapp
write_index() path does not exist rocon_apps/apps/listener/Default ros style listener tutorial
write_index() path does not exist rocon_apps/apps/listener/rocon_apps/icons/conductor.png
write_index() path does not exist rocon_apps/apps/listener/rocon_apps/apps/listener/listener.launch
write_index() attribute error : 'dict' object has no attribute 'startswith'
write_index() path does not exist rocon_apps/apps/listener/Listener
write_index() path does not exist rocon_apps/apps/listener/rocon:/
write_tarball() add .rapp file 'rocon_apps/chirp_apps/angry_cat/angry_cat_chirp.rapp
write_index() path does not exist rocon_apps/chirp_apps/angry_cat/rocon_apps/chirp
write_index() path does not exist rocon_apps/chirp_apps/angry_cat/Make an audible "angry cat" sound.
write_index() path does not exist rocon_apps/chirp_apps/angry_cat/Angry Cat Chirp
write_index() path does not exist rocon_apps/chirp_apps/angry_cat/rocon:/
write_index() path does not exist rocon_apps/chirp_apps/angry_cat/rocon_apps/chirp_apps/angry_cat/angry_cat_chirp.launch
write_tarball() add .rapp file 'rocon_apps/chirp_apps/lion/lion_chirp.rapp
write_index() path does not exist rocon_apps/chirp_apps/lion/rocon_apps/chirp
write_index() path does not exist rocon_apps/chirp_apps/lion/Make an audible "lion" sound.
write_index() path does not exist rocon_apps/chirp_apps/lion/Lion Chirp
write_index() path does not exist rocon_apps/chirp_apps/lion/rocon:/
write_index() path does not exist rocon_apps/chirp_apps/lion/rocon_apps/chirp_apps/lion/lion_chirp.launch
```

**Comment by jihoonl** [comment]
It started to happen after upgrading rapp indexer to use relative path based resource finder by https://github.com/robotics-in-concert/rocon_app_platform/pull/252

**Comment by jihoonl** [comment]
Fixed by #260
